client/src/main/java/io/cloudsoft/winrm4j/client/retry/RetryPolicy.java

```diff
@@ -0,0 +1,23 @@
+package io.cloudsoft.winrm4j.client.retry;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Policy to decide if a retry must be done.
+ */
+public interface RetryPolicy extends Function<Throwable, RetryDecision> {
+
+    /**
+     * @return total number of retries if known
+     */
+    default Optional<Integer> total() {
+        return Optional.empty();
+    }
+
+    /**
+     * Reset the state of the policy. This method MUST be called before enter the loop of retries.
+     */
+    default void clear() {

```

Comment:
I'd prefer this to be stateless. Calling `clear` makes assumptions about the threading model for how the `RetryPolicy` is used - i.e. single threaded. Given this is passed into `WinRmClientBuilder`, it is making an assumption about how the `WinRmClient` will be used.

Even if that assumption was true now, it won't necessarily be true for all users nor true in future releases.

Let's not include the `clear`, and encourage users to implement this in a thread-safe way.